analyses/survival-analysis/02-summary-statics-TP53.Rmd

```diff
@@ -293,6 +306,71 @@ ggsave(file.path(plots_dir, "boxplot_tp53_score_race.pdf"))
 
 ```
 
+### The distribution of TP53 score among predicted ancestry
+
+```{r}
+group_summary <- HGG_survival %>%
+  group_by(predicted_ancestry) %>%
+  summarize(count = n())
+
+# Filter the groups with at least 3 samples
+HGG_filtered <- HGG_survival %>%
+  filter(predicted_ancestry %in% group_summary$predicted_ancestry[group_summary$count >= 3])
+
+p1 <- HGG_filtered %>% 
+  mutate(predicted_ancestry = fct_reorder(predicted_ancestry, -tp53_score, .fun='median' )) %>%
+  ggplot(aes(x = predicted_ancestry, y = tp53_score)) + 
+  geom_boxplot() +
+  geom_jitter(width = 0.3, alpha = 0.5) + 
+  xlab("") +
+  stat_compare_means(method = "wilcox", comparisons = list(c("EAS", "AFR"), c("EAS", "EUR"), c("EAS", "AMR"), c("EAS", "Not reported"))) +
+  stat_compare_means(method = "kruskal", label.y = 1.6) + 
+  theme_Publication() + 
+  ylab("TP53 score") + 
+  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
+
+p1
```

Comment:
It looks like there is only one data point for `Not reported` ancestry, I'm assuming because some don't have TP53 scores. Maybe this should be removed in addition to SAS?